packages/iris-grid/src/mousehandlers/IrisGridContextMenuHandler.tsx

```diff
@@ -212,32 +214,36 @@ class IrisGridContextMenuHandler extends GridMouseHandler {
     } = theme;
 
     const modelSort = model.sort;
-    const columnSort = TableUtils.getSortForColumn(modelSort, modelColumn);
+    const columnSort = TableUtils.getSortForColumn(modelSort, modelIndex);
     const hasReverse = reverseType !== TableUtils.REVERSE_TYPE.NONE;
     const { userColumnWidths } = metrics;
     const isColumnHidden = [...userColumnWidths.values()].some(
       columnWidth => columnWidth === 0
     );
-    const isColumnFrozen = model.isColumnFrozen(columnIndex as VisibleIndex);
+    const isColumnFreezable =
+      model.getColumnHeaderParentGroup(modelIndex, 0) === undefined;
+    const isColumnFrozen = model.isColumnFrozen(modelIndex);
     actions.push({
       title: 'Hide Column',
       group: IrisGridContextMenuHandler.GROUP_HIDE_COLUMNS,
       action: () => {
-        this.irisGrid.hideColumnByVisibleIndex(columnIndex as VisibleIndex);
-      },
-    });
-    actions.push({
-      title: isColumnFrozen ? 'Unfreeze Column' : 'Freeze Column',
-      group: IrisGridContextMenuHandler.GROUP_HIDE_COLUMNS,
-      action: () => {
-        if (isColumnFrozen) {
-          this.irisGrid.unFreezeColumnByColumnName(column.name);
-        } else {
-          this.irisGrid.freezeColumnByColumnName(column.name);
-        }
+        this.irisGrid.hideColumnByVisibleIndex(visibleIndex);
       },
-      order: 10,
     });
+    if (isColumnFreezable) {
```

Comment:
I think instead of not showing this menu item at all, we'd want to disable it in this situation.

@dsmmcken do you have a pref?

Reply:
disable.